provider/hybrid.go

```diff
@@ -66,7 +69,139 @@ func (dp dockerHybridProvider) CheckConfig(ctx context.Context, request *rpc.Che
 
 func (dp dockerHybridProvider) DiffConfig(ctx context.Context, request *rpc.DiffRequest) (*rpc.DiffResponse, error) {
 	// Delegate to the bridged provider, as native Provider does not implement it.
-	return dp.bridgedProvider.DiffConfig(ctx, request)
+
+	urn := resource.URN(request.GetUrn())
+	label := fmt.Sprintf("DiffConfig(%s)", urn)
+
+	olds, err := plugin.UnmarshalProperties(request.GetOlds(), plugin.MarshalOptions{
+		Label:        fmt.Sprintf("%s.olds", label),
+		KeepUnknowns: true,
+		SkipNulls:    true,
+	})
+	if err != nil {
+		return nil, err
+	}
+	news, err := plugin.UnmarshalProperties(request.GetNews(), plugin.MarshalOptions{
+		Label:        fmt.Sprintf("%s.news", label),
+		KeepUnknowns: true,
+		SkipNulls:    true,
+	})
+	if err != nil {
+		return nil, fmt.Errorf("DiffConfig failed because of malformed resource inputs: %w", err)
+	}
+
+	// forcing this to true for the prototype. This needs to be configurable
+	var ignoreRegistryAuthChanges bool = true
+	if v := news["ignoreRegistryAuthChanges"]; v.HasValue() && v.IsBool() {
+		ignoreRegistryAuthChanges = v.BoolValue()
+	}
```

Comment:
I don't think we need to gate this behind provider config, FWIW. It's a bug fix IMO.

Reply:
Maybe I was a bit overcautious with that idea. My reasoning was that it might be surprising for users that a change in provider config doesn't show an update.
But you're right, that's probably only a very small subset of users (e.g. folks configuring static credentials).

FWIW The image resource just ignores changes to username&password as well without some flag.